vividus-plugin-json/src/main/java/org/vividus/json/steps/JsonExecutableSteps.java

```diff
@@ -0,0 +1,206 @@
+     *                       <li>greater than (&gt;)</li>
+     *                       <li>greater than or equal to (&gt;=)</li>
+     *                       <li>equal to (=)</li>
+     *                       <li>not equal to (!=)</li>
+     *                       </ul>
+     * @param elementsNumber The expected number of elements
+     * @param jsonPath       A JSON path
+     * @param stepsToExecute Examples table with steps to execute for each found elements
+     */
+    @SuppressWarnings("MagicNumber")
+    @When(value = "I find $comparisonRule `$elementsNumber` JSON elements by `$jsonPath` and for each element do"
+            + "$stepsToExecute", priority = 6)
+    public void performAllStepsForJsonIfFound(ComparisonRule comparisonRule, int elementsNumber, String jsonPath,
+            SubSteps stepsToExecute)
+    {
+        performAllStepsForProvidedJsonIfFound(comparisonRule, elementsNumber, getActualJson(), jsonPath,
+                stepsToExecute);
+    }
+
+    /**
+     * Step designed to perform steps against all elements found by JSON path in provided json
+     * <b>if</b> they are matching comparison rule.
+     * Actions performed by step:
+     * <ul>
+     * <li>Searches for elements using JSON path</li>
+     * <li>Checks that elements quantity matches comparison rule and elements number</li>
+     * <li>Passes if the comparison rule matches and the elements number is 0</li>
+     * <li>For each element switches JSON context and performs all steps. No steps will be performed
+     * in case of comparison rule mismatch</li>
+     * <li>Restores previously set context</li>
+     * </ul>
+     * <br> Usage example:
+     * <code>
+     * <br>When I find equal to `1` JSON elements from `{"parent_id":"","elements":[{"name": "1"},{"name": "2"}]}`
+     *       by `$.[?(@.parent_id=="")]` and for each element do
+     * <br>|step                                                      |
+     * <br>|Then number of JSON elements by JSON path `$..name` is = 2|
+     * </code>
+     * @param comparisonRule The rule to match the quantity of elements. The supported rules:
+     *                       <ul>
+     *                       <li>less than (&lt;)</li>
+     *                       <li>less than or equal to (&lt;=)</li>
+     *                       <li>greater than (&gt;)</li>
+     *                       <li>greater than or equal to (&gt;=)</li>
+     *                       <li>equal to (=)</li>
+     *                       <li>not equal to (!=)</li>
+     *                       </ul>
+     * @param elementsNumber The expected number of elements
+     * @param json           A JSON element
+     * @param jsonPath       A JSON path
+     * @param stepsToExecute Examples table with steps to execute for each found elements
+     */
+    @When("I find $comparisonRule `$elementsNumber` JSON elements from `$json` by `$jsonPath` and for each element do"
+            + "$stepsToExecute")
```

Comment:
```suggestion
    @When("I find $comparisonRule `$elementsNumber` JSON elements in `$json` by `$jsonPath` and for each element do"
            + "$stepsToExecute")
```

Reply:
It is existing step, so renaming is out of scope for this PR? :)